packages/core/src/tools/error.ts

```diff
@@ -110,3 +141,21 @@ export function createHandlingStack(): string {
 
   return formattedStack!
 }
+
+type FormatErrorCausesParams = { errorObject: ErrorWithCause; parentSource: ErrorSource }
+
+export function flattenErrorCauses({ errorObject, parentSource }: FormatErrorCausesParams): RawErrorCause[] {
+  let currentError = errorObject
+  const causes: RawErrorCause[] = []
+  while (currentError?.cause && currentError.cause && causes.length < 10) {
+    const stackTrace = currentError.cause instanceof Error ? computeStackTrace(currentError.cause) : undefined
+    causes.push({
+      message: currentError.cause.message,
+      source: parentSource,
+      type: stackTrace?.name,
+      stack: stackTrace && toStackTraceString(stackTrace),
+    })
+    currentError = currentError.cause
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏we check if `cause` is an instance of `Error` but the type says it is an `Error`.
So we then use `cause.message` and pass the `cause` as the next error regardless of it is really an `Error` or not.
Should we handle things differently if `cause` is not an `Error`?
We should probably at least have a test for it.

Reply:
I have lifted the `stackTrace` logic further up the chain and have added some logic to handle if `cause` is not of type error

Reply:
I am making the assumption that we should stop recursion if the type isnt error. IDK if this is the right assumption to make however

Reply:
> I am making the assumption that we should stop recursion if the type isnt error. IDK if this is the right assumption to make however

IMO that seems OK. We could always tweak the logic if a customer has another valid case.